File: testbed/test/cruise/associations/UnidirectionalNTest.java
package cruise.associations;

import org.junit.*;

import cruise.associations.MentorAK;
import cruise.associations.ProgramAK;
import cruise.associations.StudentAK;

public class UnidirectionalNTest
{

  @Test
  public void constructorEmpty()
  {
    MentorAK m = new MentorAK("blah");
    Assert.assertEquals(0,m.numberOfStudents());
  }
  
  @Test
  public void setTooMany()
  {
    StudentAK s = new StudentAK(99);
    StudentAK s2 = new StudentAK(98);
    StudentAK s3 = new StudentAK(97);
    StudentAK s4 = new StudentAK(96);
    
    MentorAK m = new MentorAK("blah");
    Assert.assertEquals(false,m.setStudents(s,s2,s3,s4));
    Assert.assertEquals(0, m.numberOfStudents());
  }

  
  @Test
  public void setStudents()
  {
    StudentAK s = new StudentAK(99);
    StudentAK s2 = new StudentAK(98);
    StudentAK s3 = new StudentAK(97);
    StudentAK s4 = new StudentAK(97);
    
    MentorAK m = new MentorAK("blah");
    
    Assert.assertEquals(false,m.setStudents(s,s2,s3,s4));
    Assert.assertEquals(false,m.setStudents(s,s2,s2));
    Assert.assertEquals(true,m.setStudents(s,s2));
    
    Assert.assertEquals(2,m.numberOfStudents());
  }
  
  @Test
  public void deleteDoesNotChangeStudent()
  {
    StudentAK s = new StudentAK(99);
    StudentAK s2 = new StudentAK(98);
    StudentAK s3 = new StudentAK(97);

    ProgramAK p = new ProgramAK();
    s.setProgram(p);
    
    MentorAK m = new MentorAK("blah");
    m.setStudents(s,s2,s3);
    m.delete();
    Assert.assertEquals(0,m.numberOfStudents());
    Assert.assertEquals(p,s.getProgram());
  }
}

File: testbed_python/test/associations/unidirectional_n_test.py
import unittest

# Test classes are imported dynamically
# Must be generated into ImportModules' namespace then imported
from ImportModules import importModules

importModules(["MentorAK", "ProgramAK", "StudentAK"], ["associations"])
from ImportModules import *


class UnidirectionalNTest(unittest.TestCase):
    def constructorEmpty(self):
        m = MentorAK("blah")
        self.assertEqual(0,m.numberOfStudents())
  
    def setTooMany(self):
        s = StudentAK(99)
        s2 = StudentAK(98)
        s3 = StudentAK(97)
        s4 = StudentAK(96)
        
        m = MentorAK("blah")
        self.assertEqual(False,m.setStudents(s,s2,s3,s4))
        self.assertEqual(0, m.numberOfStudents())
  
    def setStudents(self):
        s = StudentAK(99)
        s2 = StudentAK(98)
        s3 = StudentAK(97)
        s4 = StudentAK(97)
        
        m = MentorAK("blah")
        
        self.assertEqual(False,m.setStudents(s,s2,s3,s4))
        self.assertEqual(False,m.setStudents(s,s2,s2))
        self.assertEqual(True,m.setStudents(s,s2))
        
        self.assertEqual(2,m.numberOfStudents())
  
    def deleteDoesNotChangeStudent(self):
        s = StudentAK(99)
        s2 = StudentAK(98)
        s3 = StudentAK(97)

        p = ProgramAK()
        s.setProgram(p)
        
        m = MentorAK("blah")
        m.setStudents(s,s2,s3)
        del m
        self.assertEqual(0,m.numberOfStudents())
        self.assertEqual(p,s.getProgram())
  